index.js

```diff
@@ -36,12 +36,29 @@ function parse(topic) {
 	var result = {
 		regex: make_regex(tokens),
 		getParams: make_pram_getter(tokens),
-		topic: make_clean_topic(tokens)
+		topic: make_clean_topic(tokens),
+		format: make_formatter(topic)
 	};
 	result.exec = exec.bind(result);
 	return result;
 };
 
+function make_formatter(topic) {
+	function formatter(params) {
+		var tokens = tokenize(topic)
+		for (var i = 0; i < tokens.length; i++) {
```

Comment:
Would you mind refactoring this to use `.filter()` and then `.map()`?

Reply:
Refactored to use `.map()` I didn't use filter because I still want to output any tokens that aren't wildcards.